src/main/java/io/cryostat/platform/internal/KubeApiPlatformClient.java

```diff
@@ -94,62 +96,97 @@ public class KubeApiPlatformClient extends AbstractPlatformClient {
+                                                                        EventKind.FOUND,
                                                                         serviceRef));
-                                        break;
-                                    case ERROR:
-                                    default:
-                                        logger.warn(
-                                                new IllegalArgumentException(action.toString()));
-                                        return;
-                                }
-                            }
-
-                            @Override
-                            public void onClose(WatcherException watcherException) {
-                                logger.warn(watcherException);
-                            }
-                        });
+                                                break;
+                                            case ADDED:
+                                                getServiceRefs(endpoints)
+                                                        .forEach(
+                                                                serviceRef ->
+                                                                        notifyAsyncTargetDiscovery(
+                                                                                EventKind.FOUND,
+                                                                                serviceRef));
+                                                break;
+                                            case DELETED:
+                                                getServiceRefs(endpoints)
+                                                        .forEach(
+                                                                serviceRef ->
+                                                                        notifyAsyncTargetDiscovery(
+                                                                                EventKind.LOST,
+                                                                                serviceRef));
+                                                break;
+                                            case ERROR:
+                                            default:
+                                                logger.warn(
+                                                        new IllegalArgumentException(
+                                                                action.toString()));
+                                                return;
+                                        }
+                                    }
+
+                                    @Override
+                                    public void onClose(WatcherException watcherException) {
+                                        if (watcherException.isHttpGone()) {
+                                            // https://stackoverflow.com/questions/61409596/kubernetes-too-old-resource-version
```

Comment:
What about using the SharedInformer from this example instead of a Watch? Sounds like it could be less error prone.

Reply:
I can give it a try and see how it behaves.

Reply:
I just pushed a commit so you can see what I tried there. The kubernetes-client API in the version we're using is a little different from what the example in the SO post said, but I believe what I did is the equivalent. It just doesn't seem to work however - I'm not receiving any events at all.

Reply:
Fixed. I rewrote the commit history and just replaced it with the working implementation after updating the fabric8 version and replacing the Watch with an Informer.